ocaml/xapi/xapi_vdi_helpers.ml

```diff
@@ -28,6 +28,18 @@ let assert_managed ~__context ~vdi =
   if not (Db.VDI.get_managed ~__context ~self:vdi)
   then raise (Api_errors.Server_error(Api_errors.vdi_not_managed, [ Ref.string_of vdi ]))
 
+(* CA-94283: Block VDI-Destroy operation when HA Enabling/Disabling is in progress *)
+let assert_ha_in_progress ~__context ~vdi =
```

Comment:
I think this function would be better trimmed down to one that just asserts whether HA enable/disable is in progress on the pool. This function could be called when a user tries to delete an HA VDI.